build error on summit

This is an older gcc (8.1) and an old boost (1.68), but I've not seen this error before (ideas?). Using commit 
61b331f34ccf54a54f2975c975c5f156b48d0fe9 (head of master today).
I'll experiment with options and settings - I must have forgotten to set something as I wiped all my build dirs/logs from previous working versions
```
cmake -DBOOST_ROOT=$BOOST_ROOT \
 -DHPX_WITH_MALLOC=JEMALLOC -DJEMALLOC_ROOT=~/apps/gcc/8.1.1/jemalloc/5.2.1/ \
 -HWLOC_ROOT=/ccs/home/biddisco/apps/gcc/8.1.1/hwloc/2.0.4/  \
 -DHPX_WITH_MAX_CPU_COUNT=256 -DHPX_WITH_MORE_THAN_64_THREADS=ON ~/src/hpx
```
```
login4:~/build/hpx$ make mpi_ring_async_executor_test
[  0%] Built target hpx_config_registry
[  0%] Built target hpx_preprocessor
[  0%] Built target hpx_config
[  0%] Built target hpx_assertion
[  0%] Built target hpx_type_support
[  5%] Built target hpx_concepts
[  5%] Built target hpx_filesystem
[  5%] Built target hpx_format
[  5%] Built target hpx_hardware
[  5%] Built target hpx_timing
[  5%] Built target hpx_logging
[  5%] Built target hpx_thread_support
[  5%] Built target hpx_errors
[  5%] Built target hpx_basic_execution
[ 10%] Built target hpx_hashing
[ 10%] Built target hpx_concurrency
[ 10%] Built target hpx_datastructures
[ 10%] Built target hpx_debugging
[ 10%] Built target hpx_naming_base
[ 10%] Built target hpx_serialization
[ 10%] Built target hpx_functional
[ 10%] Built target hpx_plugin
[ 10%] Built target hpx_string_util
[ 10%] Built target hpx_prefix
[ 10%] Built target hpx_version
[ 10%] Built target hpx_iterator_support
[ 10%] Built target hpx_util
[ 10%] Built target hpx_topology
[ 15%] Building CXX object libs/affinity/CMakeFiles/hpx_affinity.dir/src/parse_affinity_options.cpp.o
In file included from /ccs/home/biddisco/apps/gcc/8.1.1/boost/1.68.0/include/boost/spirit/home/qi/numeric/detail/real_impl.hpp:22,
                 from /ccs/home/biddisco/apps/gcc/8.1.1/boost/1.68.0/include/boost/spirit/home/qi/numeric/real.hpp:21,
                 from /ccs/home/biddisco/apps/gcc/8.1.1/boost/1.68.0/include/boost/spirit/home/qi/numeric.hpp:17,
                 from /ccs/home/biddisco/apps/gcc/8.1.1/boost/1.68.0/include/boost/spirit/include/qi_numeric.hpp:16,
                 from /ccs/home/biddisco/src/hpx/libs/affinity/src/parse_affinity_options.cpp:23:
/ccs/home/biddisco/apps/gcc/8.1.1/boost/1.68.0/include/boost/spirit/home/support/detail/sign.hpp: In instantiation of 'T boost::spirit::detail::changesign(T) [with T = long double]':
/ccs/home/biddisco/apps/gcc/8.1.1/boost/1.68.0/include/boost/spirit/home/qi/numeric/detail/real_impl.hpp:136:50:   required from here
/ccs/home/biddisco/apps/gcc/8.1.1/boost/1.68.0/include/boost/spirit/home/support/detail/sign.hpp:60:36: error: no type named 'bits' in 'traits_type' {aka 'struct boost::math::detail::fp_traits_non_native<long double, boost::math::detail::extended_double_precision>'}
         typename traits_type::bits a;
                                    ^
/ccs/home/biddisco/apps/gcc/8.1.1/boost/1.68.0/include/boost/spirit/home/support/detail/sign.hpp:61:30: error: 'get_bits' is not a member of 'traits_type' {aka 'boost::math::detail::fp_traits_non_native<long double, boost::math::detail::extended_double_precision>'}
         traits_type::get_bits(x, a);
         ~~~~~~~~~~~~~~~~~~~~~^~~~~~
/ccs/home/biddisco/apps/gcc/8.1.1/boost/1.68.0/include/boost/spirit/home/support/detail/sign.hpp:60:36: error: no type named 'bits' in 'traits_type' {aka 'struct boost::math::detail::fp_traits_non_native<long double, boost::math::detail::extended_double_precision>'}
         typename traits_type::bits a;
                                    ^
/ccs/home/biddisco/apps/gcc/8.1.1/boost/1.68.0/include/boost/spirit/home/support/detail/sign.hpp:62:11: error: 'sign' is not a member of 'traits_type' {aka 'boost::math::detail::fp_traits_non_native<long double, boost::math::detail::extended_double_precision>'}
         a ^= traits_type::sign;
         ~~^~~~~~~~~~~~~~
/ccs/home/biddisco/apps/gcc/8.1.1/boost/1.68.0/include/boost/spirit/home/support/detail/sign.hpp:63:30: error: 'set_bits' is not a member of 'traits_type' {aka 'boost::math::detail::fp_traits_non_native<long double, boost::math::detail::extended_double_precision>'}
         traits_type::set_bits(x, a);
         ~~~~~~~~~~~~~~~~~~~~~^~~~~~
make[3]: *** [libs/affinity/CMakeFiles/hpx_affinity.dir/src/parse_affinity_options.cpp.o] Error 1
make[2]: *** [libs/affinity/CMakeFiles/hpx_affinity.dir/all] Error 2
make[1]: *** [libs/mpi/tests/unit/CMakeFiles/mpi_ring_async_executor_test.dir/rule] Error 2
make: *** [mpi_ring_async_executor_test] Error 2
```

--- a/cmake/HPX_SetupBoost.cmake
+++ b/cmake/HPX_SetupBoost.cmake
@@ -87,6 +87,13 @@ if(NOT TARGET hpx_dependencies_boost)
     )
   endif()
 
+  # target_arch should be set already by the parent CMakeLists file
+  if("${__target_arch}" STREQUAL "ppc" OR "${__target_arch}" STREQUAL "ppc64")
+    if ("${Boost_VERSION_STRING}" VERSION_LESS "1.69.0")
+      hpx_error("Boost versions < 1.69.0 are unsupported on PPC architecture")
+    endif()     
+  endif()
+
   # We are assuming that there is only one Boost Root
   if(NOT BOOST_ROOT AND "$ENV{BOOST_ROOT}")
     set(BOOST_ROOT $ENV{BOOST_ROOT})

